operator/src/test/java/org/keycloak/operator/testsuite/integration/WatchedSecretsTest.java

```diff
@@ -111,16 +150,13 @@ public void testSecretsCanBeUnWatched() {
             deployKeycloak(k8sclient, kc, false, false);
         });
 
-        Log.info("Updating DB Secret to trigger clean-up process");
-        testDeploymentRestarted(Set.of(), Set.of(kc), () -> {
-            var dbSecret = getDbSecret();
-            dbSecret.getMetadata().getLabels().put(UUID.randomUUID().toString(), "YmxhaGJsYWg");
-            k8sclient.resource(dbSecret).update();
-        });
+        Secret dbSecret = getDbSecret();
 
         Awaitility.await().untilAsserted(() -> {
             Log.info("Checking labels on DB Secret");
-            assertThat(getDbSecret().getMetadata().getLabels()).doesNotContainKey(Constants.KEYCLOAK_COMPONENT_LABEL);
+            k8sclient.resources(StatefulSet.class).withName(kc.getMetadata().getName()).get().getMetadata()
+                    .getAnnotations().get(WatchedSecretsControllerTest.KEYCLOAK_WATCHING_ANNOTATION)
+                    .contains(dbSecret.getMetadata().getName());
```

Comment:
We've removed the reference to the DB secret earlier in this test. So the annotation should NOT contain the DB secret name, right?

Reply:
Yes it should be checking for !contains.